lib/logstash/filters/split.rb

```diff
@@ -57,17 +57,15 @@ class LogStash::Filters::Split < LogStash::Filters::Base
   # The field within the new event which the value is split into.
   # If not set, the target field defaults to split field name.
   config :target, :validate => :string
-
+  
   public
   def register
     # Nothing to do
   end # def register
 
   public
   def filter(event)
-    
-
-    original_value = event.get(@field)
+    original_value = target.nil? ? event.remove(@field) : event.get(@field)
```

Comment:
Removing the field at this stage is problematic if the split operation fails, the event will be left without the field.
For example:

```
input { generator { count => 1 add_field => { "meh" => 1 } } }
filter{
  mutate { convert => { "meh" => "integer" } }
  split { field => "meh" }
}
output { stdout { codec => rubydebug } }
```

```
% bin/logstash -f cfg
[2017-07-14T10:12:38,114][WARN ][logstash.filters.split   ] Only String and Array types are splittable. field:meh is of type = NilClass
{
      "sequence" => 0,
    "@timestamp" => 2017-07-14T09:12:38.086Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!",
          "tags" => [
        [0] "_split_type_failure"
    ]
}
```